rust/xaynet-server/src/metrics/recorders/influxdb/dispatcher.rs

```diff
@@ -0,0 +1,117 @@
+use super::models::{Event, Metric};
+use derive_more::From;
+use influxdb::Client as InfluxClient;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tower::Service;
+use tracing::debug;
+
+#[derive(From)]
+pub(in crate::metrics) enum Request {
+    Metric(Metric),
+    Event(Event),
+}
+
+impl Request {
+    fn into_query(self) -> influxdb::WriteQuery {
+        match self {
+            Request::Metric(metric) => metric.into_query(),
+            Request::Event(event) => event.into_query(),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub(in crate::metrics) struct Dispatcher {
+    client: InfluxClient,
+}
+
+impl Dispatcher {
+    pub fn new(url: &str, database: &str) -> Self {
+        let client = InfluxClient::new(url, database);
+        Self { client }
+    }
+}
+
+impl Service<Request> for Dispatcher {
+    type Response = ();
+    type Error = anyhow::Error;
+    #[allow(clippy::type_complexity)]
+    type Future =
+        Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
```

Comment:
Not sure whether we want to use it but I discovered recently that `futures` provides an alias for this: https://docs.rs/futures/0.3.8/futures/future/type.BoxFuture.html

This could be `BoxFuture<'static, Result<Self::Response, Self::Error>>`

Reply:
> This could be BoxFuture<'static, Result<Self::Response, Self::Error>>

I like it, it's shorter and cleaner 👍